### ISSUE TITLE
SIGSEGV when running

### ISSUE THREAD

**Issue by lightingghost** [ethereum-mining/ethminer#1937]
**Describe the bug**
ethminer shows SIGSEGV when running

**To Reproduce**
```
/ethminer --cuda --pool stratum://xxxx.xxx@us2.ethermine.org:4444
ethminer 0.18.0
Build: darwin/release/appleclang

 i 16:17:08          Configured pool us2.ethermine.org:4444
 i 16:17:08          Selected pool us2.ethermine.org:4444
 i 16:17:08          Negotiation of EthereumStratum/2.0.0 failed. Trying another ...
 i 16:17:08          Negotiation of EthereumStratum/1.0.0 (NiceHash) failed. Trying another ...
 i 16:17:09          Stratum mode : Eth-Proxy compatible
 i 16:17:09          Established connection to us2.ethermine.org [172.65.226.101:4444]
 i 16:17:09          Spinning up miners...
cu 16:17:09 cuda-0   Using Pci Id : 01:00.0 TITAN X (Pascal) (Compute 6.1) Memory : 12.00 GB
 i 16:17:09          Epoch : 295 Difficulty : 4.00 Gh
 i 16:17:09          Job: 6e48f7a3… block 8861747 us2.ethermine.org [172.65.226.101:4444]
 i 16:17:10          Job: 26c447b7… block 8861747 us2.ethermine.org [172.65.226.101:4444]
cu 16:17:10 cuda-0   Generating DAG + Light : 3.36 GB
 i 16:17:12          Job: e36c5568… block 8861748 us2.ethermine.org [172.65.226.101:4444]
 i 16:17:12          Job: 6c754605… block 8861748 us2.ethermine.org [172.65.226.101:4444]
 m 16:17:13          0:00 A0 0.00 h - cu0 0.00
 i 16:17:17          Job: 986e20f4… block 8861748 us2.ethermine.org [172.65.226.101:4444]
 m 16:17:18          0:00 A0 0.00 h - cu0 0.00
cu 16:17:19 cuda-0   Generated DAG + Light in 8,427 ms. 8.64 GB left.
SIGSEGV encountered ...
stack trace:
backtrace() returned 8 addresses
0   ethminer                            0x000000010e104087 _ZN8MinerCLI13signalHandlerEi + 167
1   libsystem_platform.dylib            0x00007fff5f26ef5a _sigtramp + 26
2   ???                                 0x0000000119831a77 0x0 + 4722989687
3   ethminer                            0x000000010e389d1a _ZN3dev3eth9CUDAMiner8workLoopEv + 666
4   ethminer                            0x000000010e169948 _ZNSt3__114__thread_proxyINS_5tupleIJNS_10unique_ptrINS_15__thread_structENS_14default_deleteIS3_EEEEZN3dev6Worker12startWorkingEvE3$_0EEEEEPvSB_ + 168
5   libsystem_pthread.dylib             0x00007fff5f278661 _pthread_body + 340
6   libsystem_pthread.dylib             0x00007fff5f27850d _pthread_body + 0
7   libsystem_pthread.dylib             0x00007fff5f277bf9 thread_start + 13
SIGSEGV encountered ...
stack trace:
backtrace() returned 10 addresses
```

**Environment (please complete the following information):**
- Operating System: macOS 10.13.6
- Hardware Titan X Pascal
- Ethminer Version 0.18
- Ethminer options used --cuda

**Additional context**
lldb on ethminer crashed my computer

**Comment by macperlita** [comment]
Hello!
I've the same problem with a Mac Pro and GTX 1080 TI and same 0.18 ethminer version
You found a solution?
Thnx.

**Comment by simrod** [comment]
**Describe the bug**

ethminer hangs with SIGSEGV when running. ethminer runs for a few hours then hangs. Seems to fail at the same step 'cuda-0   Generating DAG + Light' as in _lightingghost_'s report.

**To Reproduce**

Command:

```
ethminer -U -P stratum2+tcp://xxxxxx@daggerhashimoto.usa.nicehash.com:3353
```

Output:

when hanging:

```
 i 22:44:02 ethminer Epoch : 376 Difficulty : 1.93 Gh                                                                                                                    
 i 22:44:02 ethminer Job: b992577b… daggerhashimoto.usa.nicehash.com [172.65.223.147:3353]                                                                               
 i 22:44:04 ethminer Extranonce set to a25c1c                                                                                                                            
 i 22:44:04 ethminer Epoch : 0 Difficulty : 1.93 Gh                                                                                                                      
 i 22:44:04 ethminer Job: f76a70f4… daggerhashimoto.usa.nicehash.com [172.65.223.147:3353]                                                                               
cu 22:44:04 cuda-0   Generating DAG + Light (reusing buffers): 4.00 GB                                                                                                   
SIGSEGV encountered ...
stack trace:
backtrace() returned 19 addresses
ethminer() [0x422af9]
/lib/x86_64-linux-gnu/libc.so.6(+0x46210) [0x7f5dc1976210]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x1d07b0) [0x7f5dbfcd07b0]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x2d03ef) [0x7f5dbfdd03ef]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x3c71bd) [0x7f5dbfec71bd]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x175a0b) [0x7f5dbfc75a0b]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x3f0885) [0x7f5dbfef0885]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x16f337) [0x7f5dbfc6f337]
/lib/x86_64-linux-gnu/libcuda.so.1(cuMemcpyHtoD_v2+0x56) [0x7f5dbfcebed6]
ethminer() [0x72ff9e]
ethminer() [0x700726]
ethminer() [0x735deb]
ethminer() [0x6e8a48]
ethminer() [0x46f265]
ethminer() [0x6eb4ec]
ethminer() [0x4baff6]
ethminer() [0x773aaf]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x9609) [0x7f5dc1c95609]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x43) [0x7f5dc1a52293]

```

after breaking with ^C:

```
^CSIGSEGV encountered ...
stack trace:
backtrace() returned 16 addresses
malloc_consolidate(): invalid chunk size
Aborted (core dumped)
```

**Environment (please complete the following information):**
 
- Operating System: Ubuntu 20.04 LTS lxc/lxd container on a Ubuntu 20.04 LTS host
- Hardware: nVidia RTX 2070
- Ethminer Version: 0.18
- Ethminer options used: -U

**Additional context:**

- CUDA 11.1 (latest)
- single GPU in system

**Comment by tanqiangyes** [comment]
same issue.

**Comment by vpolimenov** [comment]
I also get the same when running:

`./ethminer -P stratum2+tcp://xxxxxx@eu1.ethermine.org:4444 -G`

## Ubuntu version
```
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.1 LTS
Release:        20.04
Codename:       focal
```

## Nvidia Driver
```
Sat Jan  9 21:47:58 2021       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 460.32.03    Driver Version: 460.32.03    CUDA Version: 11.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  GeForce RTX 3070    Off  | 00000000:07:00.0  On |                  N/A |
|  0%   43C    P8    21W / 220W |    270MiB /  7973MiB |      1%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A      1306      G   /usr/lib/xorg/Xorg                 35MiB |
|    0   N/A  N/A      1832      G   /usr/lib/xorg/Xorg                 90MiB |
|    0   N/A  N/A      1970      G   /usr/bin/gnome-shell              132MiB |
+-----------------------------------------------------------------------------+
```

## CUDA version

```
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2020 NVIDIA Corporation
Built on Mon_Oct_12_20:09:46_PDT_2020
Cuda compilation tools, release 11.1, V11.1.105
Build cuda_11.1.TC455_06.29190527_0
```

**Comment by JordyMiggelbrink** [comment]
First all thanks to make this software available for us!

Same issue:

**Ubuntu version:**
```No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.1 LTS
Release:	20.04
Codename:	focal
```

**Nvidia driver**
```
Thu Jan 21 09:36:33 2021       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 460.27.04    Driver Version: 460.27.04    CUDA Version: 11.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  GeForce RTX 306...  On   | 00000000:01:00.0  On |                  N/A |
|100%   89C    P2   167W / 200W |   4919MiB /  7979MiB |    100%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   1  GeForce RTX 3080    On   | 00000000:02:00.0 Off |                  N/A |
| 82%   55C    P2   234W / 340W |   4427MiB / 10018MiB |    100%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A      1242      G   /usr/lib/xorg/Xorg                287MiB |
|    0   N/A  N/A      1434      G   /usr/bin/gnome-shell               85MiB |
|    0   N/A  N/A     21238      C   ./ethminer                       4335MiB |
|    1   N/A  N/A      1242      G   /usr/lib/xorg/Xorg                  4MiB |
|    1   N/A  N/A     21238      C   ./ethminer                       4419MiB |
+-----------------------------------------------------------------------------+

```
**CLI Command:**
```
./ethminer -G --report-hr -P stratum1+tcp://xxxxxxx.today@eu1.ethermine.org:4444 --farm-retries  9999
```


**Etherminer version:**
```
ethminer 0.18.0
Build: linux/release/gnu
```

**Error:**

Always occurs after a block is solved and not yet accepted by the network:

```
ethminer 0.18.0
Build: linux/release/gnu

 i 09:37:35 ethminer Configured pool eu1.ethermine.org:4444
 i 09:37:35 ethminer Selected pool eu1.ethermine.org:4444
 i 09:37:35 ethminer Stratum mode : Eth-Proxy compatible
 i 09:37:35 ethminer Established connection to eu1.ethermine.org [[2606:4700:90:0:bda0:4092:8f4f:4658]:4444]
 i 09:37:35 ethminer Spinning up miners...
cl 09:37:35 cl-0     Using PciId : 01:00.0 GeForce RTX 3060 Ti (Compute 8.6) Memory : 7.79 GB
cl 09:37:35 cl-1     Using PciId : 02:00.0 GeForce RTX 3080 (Compute 8.6) Memory : 9.78 GB
 i 09:37:35 ethminer Epoch : 389 Difficulty : 4.00 Gh
 i 09:37:35 ethminer Job: bc06f5b0… block 11697727 eu1.ethermine.org [[2606:4700:90:0:bda0:4092:8f4f:4658]:4444]
cl 09:37:37 cl-0     Generating DAG + Light : 4.10 GB
 i 09:37:37 ethminer Job: d85025f4… block 11697727 eu1.ethermine.org [[2606:4700:90:0:bda0:4092:8f4f:4658]:4444]
cl 09:37:37 cl-1     Generating DAG + Light : 4.10 GB
 i 09:37:37 ethminer Job: 64b6e40e… block 11697728 eu1.ethermine.org [[2606:4700:90:0:bda0:4092:8f4f:4658]:4444]
cl 09:37:37 cl-0     OpenCL kernel
cl 09:37:37 cl-0     Creating light cache buffer, size: 64.62 MB
cl 09:37:37 cl-0     Creating DAG buffer, size: 4.04 GB, free: 3.69 GB
cl 09:37:37 cl-0     Loading kernels
cl 09:37:37 cl-0     Writing light cache buffer
cl 09:37:37 cl-0     Creating buffer for header.
cl 09:37:37 cl-0     Creating mining buffer
cl 09:37:37 cl-1     OpenCL kernel
cl 09:37:37 cl-1     Creating light cache buffer, size: 64.62 MB
cl 09:37:37 cl-1     Creating DAG buffer, size: 4.04 GB, free: 5.68 GB
cl 09:37:37 cl-1     Loading kernels
cl 09:37:37 cl-1     Writing light cache buffer
cl 09:37:37 cl-1     Creating buffer for header.
cl 09:37:37 cl-1     Creating mining buffer
 i 09:37:37 ethminer Job: 85b103b3… block 11697728 eu1.ethermine.org [[2606:4700:90:0:bda0:4092:8f4f:4658]:4444]
 m 09:37:40 ethminer 0:00 A0 0.00 h - cl0 0.00, cl1 0.00
 i 09:37:41 ethminer Job: 071f9830… block 11697728 eu1.ethermine.org [[2606:4700:90:0:bda0:4092:8f4f:4658]:4444]
cl 09:37:43 cl-1     4.04 GB of DAG data generated in 5.654 ms.
cl 09:37:45 cl-0     4.04 GB of DAG data generated in 7.698 ms.
 i 09:37:45 ethminer Job: f9c67912… block 11697728 eu1.ethermine.org [[2606:4700:90:0:bda0:4092:8f4f:4658]:4444]
 i 09:37:53 ethminer Job: 8cc14fe2… block 11697729 eu1.ethermine.org [[2606:4700:90:0:bda0:4092:8f4f:4658]:4444]
 m 09:38:00 ethminer 0:00 A0 0.00 h - cl0 0.00, cl1 0.00
 i 09:38:00 ethminer Job: 89993784… block 11697729 eu1.ethermine.org [[2606:4700:90:0:bda0:4092:8f4f:4658]:4444]
 i 09:38:01 ethminer Job: a6f77f06… block 11697730 eu1.ethermine.org [[2606:4700:90:0:bda0:4092:8f4f:4658]:4444]
cl 09:38:02 cl-1     Job: a6f77f06… Sol: 0xe019e93ca93640d4
 i 09:38:02 ethminer Job: 8730dd7a… block 11697731 eu1.ethermine.org [[2606:4700:90:0:bda0:4092:8f4f:4658]:4444]
 X 09:38:04 ethminer No response received in 2 seconds.
 i 09:38:04 ethminer **Accepted   0 ms. eu1.ethermine.org [[2606:4700:90:0:bda0:4092:8f4f:4658]:4444]
 i 09:38:04 ethminer Job: 8bf68439… block 11697731 eu1.ethermine.org [[2606:4700:90:0:bda0:4092:8f4f:4658]:4444]
 i 09:38:05 ethminer Disconnected from eu1.ethermine.org [[2606:4700:90:0:bda0:4092:8f4f:4658]:4444]
 i 09:38:05 ethminer No connection. Suspend mining ...
 i 09:38:05 ethminer Selected pool eu1.ethermine.org:4444
 m 09:38:05 ethminer Not connected
 i 09:38:05 ethminer Stratum mode : Eth-Proxy compatible
 i 09:38:05 ethminer Established connection to eu1.ethermine.org [[2606:4700:90:0:bda0:4092:8f4f:4658]:4444]
 i 09:38:05 ethminer Resume mining ...
 i 09:38:05 ethminer Job: 8bf68439… block 11697731 eu1.ethermine.org [[2606:4700:90:0:bda0:4092:8f4f:4658]:4444]
 i 09:38:08 ethminer Job: 692e1831… block 11697731 eu1.ethermine.org [[2606:4700:90:0:bda0:4092:8f4f:4658]:4444]
 i 09:38:27 ethminer Job: e2c1a216… block 11697731 eu1.ethermine.org [[2606:4700:90:0:bda0:4092:8f4f:4658]:4444]
cl 09:38:37 cl-0     Job: e2c1a216… Sol: 0xe019e93bc55cb5b7
 m 09:38:39 ethminer 0:01 A1 128.38 Mh - cl0 45.30, cl1 83.08
 i 09:38:39 ethminer Job: e81ca01c… block 11697732 eu1.ethermine.org [[2606:4700:90:0:bda0:4092:8f4f:4658]:4444]
 i 09:38:49 ethminer Job: 7a7c848e… block 11697732 eu1.ethermine.org [[2606:4700:90:0:bda0:4092:8f4f:4658]:4444]
cl 09:38:53 cl-1     Job: 7a7c848e… Sol: 0xe019e93cac9c2059
cl 09:38:55 cl-1     Job: 7a7c848e… Sol: 0xe019e93cb934b618
 X 09:38:55 ethminer No response received in 2 seconds.
 m 09:38:55 ethminer 0:01 A1 135.87 Mh - cl0 50.09, cl1 85.78
 i 09:38:55 ethminer Job: ea66173d… block 11697733 eu1.ethermine.org [[2606:4700:90:0:bda0:4092:8f4f:4658]:4444]
 i 09:39:02 ethminer Disconnected from eu1.ethermine.org [[2606:4700:90:0:bda0:4092:8f4f:4658]:4444]
 i 09:39:02 ethminer No connection. Suspend mining ...
SIGSEGV encountered ...
stack trace:
backtrace() returned 13 addresses
./ethminer() [0x422af9]
/lib/x86_64-linux-gnu/libc.so.6(+0x46210) [0x7fd4d7a35210]
./ethminer() [0x4956b8]
./ethminer() [0x49a016]
./ethminer() [0x48d03e]
./ethminer() [0x48eda9]
./ethminer() [0x48eff7]
./ethminer() [0x492071]
./ethminer() [0x4922cd]
./ethminer() [0x42a2d0]
./ethminer() [0x773aaf]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x9609) [0x7fd4d7d54609]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x43) [0x7fd4d7b11293]
```

**Comment by simrod** [comment]
**Describe the bug**
 
ethminer hangs with SIGSEGV when running. ethminer runs for a few hours then hangs. Seems to fail at the same step 'cuda-0 Generating DAG + Light' as in _lightingghost_'s report.

I have compiled the current master: commit 47ae149ee734a82c453d23ef46fed78da109b73e to pick up any changes since my first report which used version 0.18.0.

Note that I continue to have the problem when trying to mine at nicehash, but it appears to work at ethermine.com w/o incident.

**To Reproduce** 
Command:

```
ethminer -U -P stratum2+tcp://xxxxxx@daggerhashimoto.usa.nicehash.com:3353
```

Output:

when hanging:
 
```
i 22:44:02 ethminer Epoch : 376 Difficulty : 1.93 Gh                                                                                                                    
 i 22:44:02 ethminer Job: b992577b… daggerhashimoto.usa.nicehash.com [172.65.223.147:3353]                                                                               
 i 22:44:04 ethminer Extranonce set to a25c1c                                                                                                                            
 i 22:44:04 ethminer Epoch : 0 Difficulty : 1.93 Gh                                                                                                                      
 i 22:44:04 ethminer Job: f76a70f4… daggerhashimoto.usa.nicehash.com [172.65.223.147:3353]                                                                               
cu 22:44:04 cuda-0   Generating DAG + Light (reusing buffers): 4.00 GB                                                                                                   
SIGSEGV encountered ...
stack trace:
backtrace() returned 19 addresses
ethminer() [0x422af9]
/lib/x86_64-linux-gnu/libc.so.6(+0x46210) [0x7f5dc1976210]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x1d07b0) [0x7f5dbfcd07b0]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x2d03ef) [0x7f5dbfdd03ef]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x3c71bd) [0x7f5dbfec71bd]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x175a0b) [0x7f5dbfc75a0b]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x3f0885) [0x7f5dbfef0885]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x16f337) [0x7f5dbfc6f337]
/lib/x86_64-linux-gnu/libcuda.so.1(cuMemcpyHtoD_v2+0x56) [0x7f5dbfcebed6]
ethminer() [0x72ff9e]
ethminer() [0x700726]
ethminer() [0x735deb]
ethminer() [0x6e8a48]
ethminer() [0x46f265]
ethminer() [0x6eb4ec]
ethminer() [0x4baff6]
ethminer() [0x773aaf]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x9609) [0x7f5dc1c95609]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x43) [0x7f5dc1a52293]
```

after breaking with ^C:

```
^CSIGSEGV encountered ...
stack trace:
backtrace() returned 16 addresses
malloc_consolidate(): invalid chunk size
Aborted (core dumped)
```

**Environment (please complete the following information):**

* Operating System: Ubuntu 20.04.1 LTS lxc/lxd container on a Ubuntu 21.10 lxd host
* Hardware: nVidia RTX 2070, RTX 3090
* Ethminer Version: 0.19.0-16+commit.47ae149e
* git commit: 47ae149ee734a82c453d23ef46fed78da109b73e
* Ethminer options used: -U
 
'''
 Thu Jan 21 10:04:09 2021       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 460.32.03    Driver Version: 460.32.03    CUDA Version: 11.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  GeForce RTX 2070    Off  | 00000000:01:00.0 Off |                  N/A |
| 49%   64C    P2   125W / 125W |   4308MiB /  7982MiB |    100%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   1  GeForce RTX 3090    Off  | 00000000:02:00.0 Off |                  N/A |
| 95%   61C    P2   219W / 220W |   4468MiB / 24268MiB |    100%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A   1422338      C   ...r/build/ethminer/ethminer     4305MiB |
|    1   N/A  N/A   1422338      C   ...r/build/ethminer/ethminer     4465MiB |
+-----------------------------------------------------------------------------+
'''

**Additional context:**

* CUDA 11.2 (latest)
* two GPUs in system

**Comment by dkgeorge** [comment]
I'm running two RTX 3070 (8GB) and what I've observed is that I cannot start GPU 0 before GPU 1 without a DAG size error. 

`
ethminer 0.19.0-16+commit.47ae149e
Build: linux/release/gnu

 m 08:32:31 ethminer DBus initialized!
 i 08:32:32 ethminer Configured pool daggerhashimoto.usa.nicehash.com:3353
 i 08:32:32 ethminer Selected pool daggerhashimoto.usa.nicehash.com:3353
 i 08:32:32 ethminer Stratum mode : EthereumStratum/1.0.0 (NiceHash)
 i 08:32:32 ethminer Established connection to daggerhashimoto.usa.nicehash.com [172.65.223.147:3353]
 i 08:32:32 ethminer Spinning up miners...
cu 08:32:32 cuda-0   Using Pci Id : 02:00.0 GeForce RTX 3070 (Compute 8.6) Memory : 3.35 GB
 i 08:32:32 ethminer Extranonce set to 509e9c
 i 08:32:32 ethminer Authorized worker xxxxxxxxxxxxxxxxxxxxxxxxxx.RTX3070_1
 i 08:32:32 ethminer Epoch : 394 Difficulty : 1.67 Gh
 i 08:32:32 ethminer Job: 5cc033d8… daggerhashimoto.usa.nicehash.com [172.65.223.147:3353]
 i 08:32:35 ethminer Job: ea9992d6… daggerhashimoto.usa.nicehash.com [172.65.223.147:3353]
cu 08:32:35 cuda-0   Epoch 394 requires 4.08 GB memory.
cu 08:32:35 cuda-0   This device hasn't enough memory available. Mining suspended ...
 i 08:32:36 ethminer Job: 0f6dbce0… daggerhashimoto.usa.nicehash.com [172.65.223.147:3353]
 m 08:32:37 ethminer 0:00 A0 0.00 h - cu0 0.00 52C 30% 44.95W
 i 08:32:39 ethminer Job: eb0b70e3… daggerhashimoto.usa.nicehash.com [172.65.223.147:3353]
 m 08:32:42 ethminer 0:00 A0 0.00 h - cu0 0.00 52C 30% 44.98W
 i 08:32:42 ethminer Job: 4872b347… daggerhashimoto.usa.nicehash.com [172.65.223.147:3353]
^C i 08:32:46 main     Got interrupt ...
 i 08:32:46 main     Disconnected from daggerhashimoto.usa.nicehash.com [172.65.223.147:3353]
 i 08:32:46 main     Shutting down miners...
 i 08:32:46 main     Terminated!
`

To get around it I have to start the ethminer instance for GPU 1 first and then the other for GPU 0. 
After doing this, it runs fine for a some hours and then crashes.

`
cu 18:20:42 cuda-0   Job: f5b07209… Sol: 0xda5a2b001527af7d
 i 18:20:42 ethminer **Accepted  58 ms. daggerhashimoto.usa.nicehash.com [172.65.223.147:3353]
 i 18:20:43 ethminer Extranonce set to dbbbc3
 i 18:20:43 ethminer Epoch : 0 Difficulty : 644.25 Mh
 i 18:20:43 ethminer Job: f5b07209… daggerhashimoto.usa.nicehash.com [172.65.223.147:3353]
 i 18:20:43 ethminer Epoch : 393 Difficulty : 644.25 Mh
 i 18:20:43 ethminer Job: 2ca68c2c… daggerhashimoto.usa.nicehash.com [172.65.223.147:3353]
cu 18:20:44 cuda-0   Job: f5b07209… Sol: 0xdbbbc30003606cb2
cu 18:20:45 cuda-0   Generating DAG + Light (reusing buffers): 4.13 GB
SIGSEGV encountered ...
stack trace:
backtrace() returned 19 addresses
build/ethminer/ethminer(+0x99a30) [0x5639f586fa30]
/lib/x86_64-linux-gnu/libc.so.6(+0x3bd60) [0x7f963069bd60]
/usr/lib/x86_64-linux-gnu/libcuda.so.1(+0x1d95c0) [0x7f962e3195c0]
/usr/lib/x86_64-linux-gnu/libcuda.so.1(+0x31c553) [0x7f962e45c553]
/usr/lib/x86_64-linux-gnu/libcuda.so.1(+0x405885) [0x7f962e545885]
/usr/lib/x86_64-linux-gnu/libcuda.so.1(+0x176eac) [0x7f962e2b6eac]
/usr/lib/x86_64-linux-gnu/libcuda.so.1(+0x3e85d5) [0x7f962e5285d5]
/usr/lib/x86_64-linux-gnu/libcuda.so.1(+0x16f3b7) [0x7f962e2af3b7]
/usr/lib/x86_64-linux-gnu/libcuda.so.1(cuMemcpyHtoD_v2+0x56) [0x7f962e369156]
build/ethminer/ethminer(+0x38dbb9) [0x5639f5b63bb9]
build/ethminer/ethminer(+0x365235) [0x5639f5b3b235]
build/ethminer/ethminer(+0x3ab149) [0x5639f5b81149]
build/ethminer/ethminer(+0x34c700) [0x5639f5b22700]
build/ethminer/ethminer(+0xe31f7) [0x5639f58b91f7]
build/ethminer/ethminer(+0x34f038) [0x5639f5b25038]
build/ethminer/ethminer(+0x1294e7) [0x5639f58ff4e7]
build/ethminer/ethminer(+0x4545ef) [0x5639f5c2a5ef]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x8ea7) [0x7f963098dea7]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x3f) [0x7f963075ddef]

SIGSEGV encountered ...
stack trace:
backtrace() returned 16 addresses
malloc_consolidate(): invalid chunk size
`

I can confirm that versions up v0.17.1 are not affected by this issue. 
Below is the building  modification I used to enable support for RTX 3070 in v0.17.1:

```
$ git clone https://github.com/ethereum-mining/ethminer.git -b release/0.17 && cd ethminer && \
git submodule update --init --recursive && mkdir -p build && cd build && cmake .. -DETHASHCUDA=ON -DETHDBUS=ON -DETHASHCL=OFF && \
sed -i 's/arch=compute_30,code=sm_30;-gencode arch=compute_35,code=sm_35;-gencode arch=compute_50,code=sm_50;-gencode arch=compute_52,code=sm_52;-gencode arch=compute_53,code=sm_53;-gencode arch=compute_60,code=sm_60;-gencode arch=compute_61,code=sm_61;-gencode arch=compute_62,code=sm_62;-gencode arch=compute_70,code=sm_70;-gencode arch=compute_75,code=sm_75/arch=compute_35,code=sm_35;-gencode arch=compute_50,code=sm_50;-gencode arch=compute_52,code=sm_52;-gencode arch=compute_53,code=sm_53;-gencode arch=compute_60,code=sm_60;-gencode arch=compute_61,code=sm_61;-gencode arch=compute_62,code=sm_62;-gencode arch=compute_70,code=sm_70;-gencode arch=compute_75,code=sm_75;-gencode arch=compute_80,code=sm_80;-gencode arch=compute_86,code=sm_86/g' libethash-cuda/CMakeFiles/ethash-cuda.dir/ethash-cuda_generated_ethash_cuda_miner_kernel.cu.o.Release.cmake && \
cmake --build .
```

**Comment by tarfeef101** [comment]
I know this is not a solution, but hopefully this little script helps you keep your rigs active. I just set a cron job to run this every minute:

```
$!/bin/bash
util=$(nvidia-smi -q -d UTILIZATION | awk '/Gpu/ { print $3 }')
if [ $util -lt 80 ]; then
  echo "restarting container"
  cd /location/of/config
  /usr/local/bin/docker-compose restart
else
  echo "continuing"
fi
```

if you're not running in docker, then you totally should :D. but if not, just replace the `cd` and `restart` with killing and starting the binary again. 

I have a gist with the full setup I use if anyone is interested [here](https://gist.github.com/tarfeef101/1721dfabf0a2d6c559925b0b9f2d31a0).

**Comment by nerdcorenet** [comment]
This issue and #2258 both appear to be duplicates of #1895

**Comment by algo7** [comment]
Same here
